reccmp/isledecomp/cvdump/types.py

```diff
@@ -442,84 +503,79 @@ def read_all(self, section: str):
 
             (leaf_id, leaf_type) = match.groups()
             if leaf_type not in self.MODES_OF_INTEREST:
-                self.mode = None
                 continue
 
-            # Add the leaf to our dictionary and add details specific to the leaf type.
-            self.mode = leaf_type
-            self.keys[leaf_id] = {"type": leaf_type}
-
-            this_key = self.keys[leaf_id]
-
             try:
-                if self.mode == "LF_MODIFIER":
-                    this_key.update(self.read_modifier(leaf))
+                if leaf_type == "LF_MODIFIER":
```

Comment:
Drive-by: Can this be refactored to `match-case` now that we have a minimum of Python 3.10?